src/ui/pages/BridgeForm/BridgeForm.tsx

```diff
@@ -0,0 +1,1172 @@
+    enabled:
+      quotesData.done &&
+      Boolean(selectedQuote && !selectedQuote.enough_allowance),
+    keepPreviousData: false,
+  });
+
+  const sendTxBtnRef = useRef<SendTxBtnHandle | null>(null);
+  const approveTxBtnRef = useRef<SendTxBtnHandle | null>(null);
+
+  const txToCheck =
+    quotesData.done && !enough_allowance
+      ? approvalTransaction
+      : bridgeTransaction;
+
+  const { data: networkNonce, refetch: refetchNonce } = useQuery({
+    suspense: false,
+    queryKey: ['uiGetBestKnownTransactionCount', address, spendChain, networks],
+    queryFn: async () => {
+      if (!spendChain || !networks) {
+        return null;
+      }
+      const result = await uiGetBestKnownTransactionCount({
+        address,
+        chain: spendChain,
+        networks,
+        defaultBlock: 'pending',
+      });
+      return result.value;
+    },
+  });
+
+  const [txConfiguration, setTxConfiguration] = useState(DEFAULT_CONFIGURATION);
+
+  const USE_PAYMASTER_FEATURE = true;
+
+  const userNonce = txConfiguration.nonce;
+  const nonce = userNonce ?? networkNonce ?? undefined;
+  const gas = txToCheck ? getGas(txToCheck) : null;
+
+  const network = spendChain ? networks?.getNetworkByName(spendChain) : null;
+
+  const eligibilityParams:
+    | null
+    | Parameters<ZerionApiClient['paymasterCheckEligibility']>[0] =
+    USE_PAYMASTER_FEATURE && txToCheck && txToCheck.to && nonce != null && gas
+      ? {
+          chainId: txToCheck.chainId,
+          from: address,
+          to: txToCheck.to,
+          value: 'value' in txToCheck ? txToCheck.value : null,
+          data: txToCheck.data,
+          gas,
+          nonce: Number(nonce),
+        }
+      : null;
+  const paymasterPossible =
+    USE_PAYMASTER_FEATURE && Boolean(network?.supports_sponsored_transactions);
+  const eligibilityQuery = useTxEligibility(eligibilityParams, {
+    enabled: paymasterPossible,
+  });
+
+  const paymasterEligible = Boolean(eligibilityQuery.data?.data.eligible);
+
+  const { data: gasbackEstimation } = useGasbackEstimation({
+    paymasterEligible: eligibilityQuery.data?.data.eligible ?? null,
+    suppportsSimulations: network?.supports_simulations ?? false,
+    supportsSponsoredTransactions: network?.supports_sponsored_transactions,
+  });
+
+  const configureTransactionToBeSigned = useEvent(
+    (tx: IncomingTransactionWithChainId) => {
+      invariant(spendChain && networks, 'Not ready to prepare the transaction');
+      let txToSign = applyConfiguration(tx, txConfiguration, gasPrices);
+      if (paymasterEligible && txToSign.nonce == null) {
+        txToSign = {
+          ...txToSign,
+          nonce: nonce != null ? Number(nonce) : undefined,
+        };
+        invariant(txToSign.nonce, 'Nonce is required for a paymaster tx');
+      }
+      return { ...txToSign, from: address };
+    }
+  );
+
+  const { preferences } = usePreferences();
+  const source = preferences?.testnetMode?.on ? 'testnet' : 'mainnet';
```

Comment:
Testnet mode is not supported for bridge. I think, we can skip these variations inside bridge form code

Reply:
I think for correctness these conditionals can remain. We have the same in SwapForm